toolsrc/src/vcpkg/commands.porthistory.cpp

```diff
@@ -27,24 +27,27 @@ namespace vcpkg::Commands::PortHistory
             int port_version;
         };
 
-        const System::ExitCodeAndOutput run_git_command_inner(const VcpkgPaths& paths,
-                                                              const fs::path& dot_git_directory,
-                                                              const fs::path& working_directory,
-                                                              const std::string& cmd)
+        System::ExitCodeAndOutput run_git_command_inner(const VcpkgPaths& paths,
+                                                        const fs::path& dot_git_directory,
+                                                        const fs::path& working_directory,
+                                                        StringView cmd)
         {
             const fs::path& git_exe = paths.get_tool_exe(Tools::GIT);
 
             System::CmdLineBuilder builder;
             builder.path_arg(git_exe)
                 .string_arg(Strings::concat("--git-dir=", fs::u8string(dot_git_directory)))
                 .string_arg(Strings::concat("--work-tree=", fs::u8string(working_directory)));
-            const std::string full_cmd = Strings::concat(builder.extract(), " ", cmd);
 
-            const auto output = System::cmd_execute_and_capture_output(full_cmd);
+            auto full_cmd = std::move(builder).extract();
+            full_cmd.append(" ");
+            full_cmd.append(cmd.begin(), cmd.end());
```

Comment:
`Strings::concat()` is optimized to reuse an rvalue string as the first parameter, so there's no need to write this out long-form.

```suggestion
            const auto full_cmd = Strings::concat(std::move(builder).extract(), " ", cmd);
```